extensions/src/test/java/org/pentaho/platform/plugin/services/email/EmailServiceTest.java

```diff
@@ -0,0 +1,285 @@
+/*!
+ *
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ *
+ * Copyright (c) 2002-2023 Hitachi Vantara. All rights reserved.
+ *
+ */
+
+package org.pentaho.platform.plugin.services.email;
+
+import junit.framework.TestCase;
+import org.apache.commons.lang.ObjectUtils;
+import org.apache.http.HttpException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.pentaho.di.core.encryption.Encr;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Properties;
+
+@RunWith( MockitoJUnitRunner.class )
+public class EmailServiceTest extends TestCase {
+
+  private EmailService emailService = null;
+
+  private File defaultConfigFile = null;
+  @Before
+  public void setUp() throws Exception {
+    defaultConfigFile = File.createTempFile( "email_config_", ".xml" );
+    this.emailService = new EmailService( defaultConfigFile );
+  }
+
+
+  @Test( expected = HttpException.class )
```

Comment:
This function `EmailService#sendMailGraphApi(` sends off an Http Post to a URL with certain headers and body contents. and then checks if Http status is 202. I don't see non of that verification in the unit test.

You should be using testing frameworks like

- wire mock
- - https://www.baeldung.com/introduction-to-wiremock
- mock server
-- https://github.com/square/okhttp/tree/master/mockwebserver